routes/api/contacts.js

```diff
@@ -1,25 +1,143 @@
+import express from 'express';
 
-const router = express.Router()
+import {
+  listContacts,
+  getContactById,
+  removeContact,
+  addContact,
+  updateContact,
+  updatedStatusContact,
+} from './../../models/contacts.js';
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+import { auth } from '../../config/config-passport.js';
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+export const contactsRouter = express.Router();
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+const paginatedResults = (array, page, limit) => {
+  const startIndex = (page - 1) * limit;
+  const endIndex = page * limit;
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+  const results = array.slice(startIndex, endIndex);
+  return results;
+};
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+contactsRouter.get('/', auth, async (req, res, next) => {
+  try {
+    const { id: userID } = req.user;
+    const contacts = await listContacts(userID);
+    const { page = 1, limit = 20, favorite } = req.query;
+    let filteredContacts = contacts;
+    if (favorite === 'true') {
+      filteredContacts = contacts.filter(contact => contact.favorite);
+    }
+    const paginatedContacts = paginatedResults(filteredContacts, page, limit);
 
-module.exports = router
+    return res.status(200).json({
+      status: 'success',
+      code: 200,
+      data: { contacts: paginatedContacts },
+    });
+  } catch (error) {
+    res.status(500).json({
+      status: 'error',
+      code: 500,
+      message: `An error occurred while getting the contact list: ${err}`,
+    });
+  }
+});
+
+contactsRouter.get('/:id', auth, async (req, res, next) => {
+  const { id: userId } = req.user;
+  const { id: contactId } = req.params;
+  try {
+    const contact = await getContactById(userId, contactId);
+
+    return res.status(200).json({
+      status: 'success',
+      code: 200,
+      data: { contact },
+    });
+  } catch (err) {
+    res.status(500).json(`An error occurred while getting the contact: ${err}`);
+  }
+});
+
+contactsRouter.post('/', auth, async (req, res, next) => {
+  const body = req.body;
+  const { id: userId } = req.user;
+
+  if (Object.keys(body).length === 0) {
+    return res.status(400).json('Error! Missing fields! Empty request is not allowed');
+  }
+
+  try {
+    const contact = await addContact(body, userId);
+
+    return res.status(201).json({
+      status: 'success',
+      code: 201,
+      data: { contact },
+    });
+  } catch (err) {
+    res.status(500).json(`An error occurred while adding the contact: ${err}`);
+  }
+});
+
+contactsRouter.delete('/:id', auth, async (req, res, next) => {
+  const { id: userId } = req.user;
+  const { id: contactId } = req.params;
```

Comment:
if you query by mongoose `_id` its good idea to validate it with `mongoose.isValidObjectId`